static/panes/compiler.ts

```diff
@@ -388,6 +378,23 @@ export class Compiler extends MonacoPane<monaco.editor.IStandaloneCodeEditor, Co
         }
     }
 
+    override initializeStateDependentProperties(state: MonacoPaneState & CompilerState) {
+        this.compilerService = this.hub.compilerService;
+        this.sourceTreeId = state.tree ? state.tree : null;
+        if (this.sourceTreeId) {
+            this.sourceEditorId = null;
+        } else {
+            this.sourceEditorId = state.source || 1;
+        }
+        this.options = state.options || (options.compileOptions[this.currentLangId ?? ''] ?? '');
+
+        this.deviceViewOpen = !!state.deviceViewOpen;
```

Comment:
Please do not merge until I test that the device and selection still work. I can't test device without using staging

Reply:
you can test it locally if you have Clang, ICX or NVCC

Reply:
device viewer still works